index.rst

```diff
@@ -437,284 +329,111 @@ this case, we may have something like:
+This section describes some of the issues we found and propose changes to the system to improve the user experience and system reliability.
 
-Use of "recommended" in settingVersions event
----------------------------------------------
+The following should be seen as an open floor for discussions and we expect developers and users to comment and provide feedback before we can start implementation.
+It should also be noted that these changes will require work from Telescope and Site and other sub-systems.
+For components written in Salobj it should be straightforward to implement these changes but those :ref:`handcrafted CSCs <section-handcrafted>` will need to be updated case by case.
 
-The use of "recommended" in both the main attributes of ``settingVersions``
-event is a bit of a misnomer. The ``recommendedSettingsLabels`` actually
-specify available labels and ``recommendedSettingsVersion`` specify the version
-of the available configuration.
+.. _section-renaming:
 
-One suggestion would be to rename those to ``labels`` and ``version`` only,
-see :ref:`section-appendix-renaming` furthermore.
+Topics/attributes renaming
+--------------------------
 
-.. _section-appendix-mapping:
+The interface would be much clear if we rename some generic topics and attributes to reflect more closely their true meaning.
 
-Configuration mapping
----------------------
+The following is a renaming suggestion for discussion:
 
-The mapping between label, configuration file, and branch description is not
-sufficiently straightforward, especially when trying to analyse configurations
-used in the past.
-
-For instance, to perform a full reconstruction using the current framework on
-a salobj CSC, the user needs to know the configuration repository (which is
-specified in ``settingVersions.settingsUrl``), the label used to configure the
-CSC (specified in ``start.settingsToApply`` command) and the version of the
-configuration (specified in both ``settingVersions.recommendedSettingsVersion``
-and ``settingsApplied.settingsVersion``). To get the label to configuration
-file mapping the user then need to go to the configuration repository, find the
-version that matches the one used by the CSC at that time, check the
-``_label.yaml`` file so they can finally find out what configuration was used.
-
-This information should probably be present in both ``settingVersions`` and
-``settingsApplied``. For instance ``settingVersions`` should contain an
-additional entry, e.g. ``mapping``, with the names of the files the labels
-maps to (in case of git-based configuration) or any other appropriate mapping
-information (e.g. for database configuration). Then, ``settingsApplied`` should
-also have an entry for ``label`` and ``mapping`` that will contain information
-about the selected label.
-
-.. _section-appendix-available-vs-selected:
-
-Available vs. selected
-----------------------
+#.  Rename ``settingsVersions`` to ``configurationsAvailable``
```

Comment:
Add a sentence saying something like "This topic presents all the available configurations that can be loaded by the CSC"